utils/build/docker/dotnet/parametric/Services/ApmTestClientService.Otel.cs

```diff
@@ -0,0 +1,245 @@
+                    parentSpanId,
+                    flags,
+                    extractedContext.AdditionalW3CTraceState,
+                    isRemote: true);
+            }
+        }
+
+        // sanity check that we didn't receive both a local and remote parent
+        if (localParentContext != null && remoteParentContext != null)
+        {
+            throw new ApplicationException(
+                "Both ParentId and HttpHeaders were provided to OtelStartSpan(). " +
+                "Provide one or the other, but not both.");
+        }
+
+        var startTime = request.HasTimestamp ? DateTimeOffset.FromUnixTimeMilliseconds(request.Timestamp) : default;
+        var parentContext = localParentContext ?? remoteParentContext ?? default;
+
+        var activity = ApmTestClientActivitySource.StartActivity(
+            request.Name,
+            ActivityKind.Internal,
+            parentContext,
+            tags: null,
+            links: null,
+            startTime);
+
+        if (activity is null)
+        {
+            throw new ApplicationException("Failed to start activity. Make sure there are listeners registered.");
+        }
+
+        activity.ActivityTraceFlags = ActivityTraceFlags.Recorded;
+
+        _logger.LogInformation("Started Activity: OperationName={OperationName}", activity.OperationName);
+
+        ulong traceId = ulong.Parse(activity.TraceId.ToString().AsSpan(16, 16), NumberStyles.HexNumber);
+        ulong spanId = ulong.Parse(activity.SpanId.ToString(), NumberStyles.HexNumber);
+
+        if (!Activities.TryAdd(spanId, activity))
+        {
+            throw new ApplicationException("Failed to add activity to dictionary.");
+        }
+
+        var result = new OtelStartSpanReturn
+                     {
+                         TraceId = traceId,
+                         SpanId = spanId
+                     };
+
+        _logger.LogInformation("OtelStartSpanReturn: {Result}", result);
+        return Task.FromResult(result);
+    }
+
+    public override Task<OtelEndSpanReturn> OtelEndSpan(OtelEndSpanArgs request, ServerCallContext context)
+    {
+        _logger.LogInformation("OtelEndSpan: {Request}", request);
+
+        var activity = FindActivity(request.Id);
+        activity.Stop();
+
+        if (request.HasTimestamp)
+        {
+            DateTimeOffset timestamp = DateTimeOffset.FromUnixTimeMilliseconds(request.Timestamp);
+            activity.SetEndTime(timestamp.DateTime);
+        }
+
+        _logger.LogInformation("OtelEndSpanReturn");
+        return Task.FromResult(new OtelEndSpanReturn());
+    }
+
+    public override Task<OtelIsRecordingReturn> OtelIsRecording(OtelIsRecordingArgs request, ServerCallContext context)
+    {
+        _logger.LogInformation("OtelIsRecording: {Request}", request);
+
+        var activity = FindActivity(request.SpanId);
+
+        var result = new OtelIsRecordingReturn
+                     {
+                         IsRecording = activity.Recorded
```

Comment:
Though not really documented anywhere, the translation from OTEL Spans to Activities is Span.IsRecording => Activity.IsAllDataRequested

See https://github.com/open-telemetry/opentelemetry-dotnet/issues/947

```suggestion
                         IsRecording = activity.IsAllDataRequested
```

Reply:
Whoa, thanks! I will re-run this tests and unskip if possible before merging.

Reply:
While I confirmed that this is the correct property by reading the source code, it still suffers from the same issue: `Activity.IsAllDataRequested` is not set to `false` when the activity ends.

Reply:
Yeah, that bug is still present, and who knows if it will ever be fixed. The behavior is documented as a _should_ in the spec not a _must_ 🤷🏼